src/core/lib/channel/metrics.h

```diff
@@ -110,6 +138,24 @@ class GlobalInstrumentsRegistry {
   GetInstrumentList();
 };
 
+// An interface for implementing callback-style metrics.
+// To be implemented by stats plugins.
+class CallbackMetricReporter {
```

Comment:
Why not move this into the `StatsPlugin` interface as a nested class? Since the `CallbackmetricReporter` probably needs to access members within the `StatsPlugin` to access the asynchronous instrument and making it a nested class may make that easier.

Reply:
A stats plugin that implements this interface can have that implementation nested within the stats plugin subclass.  The base class does not access any data, so there's no advantage of it being inside `StatsPlugin`.

Note that this interface is also needed by the gRPC plugin that generates the metric data, and that plugin does not otherwise need to use `StatsPlugin` directly at all; it only interacts with `StatsPluginGroup`.  So that seems to be a good reason not to put this inside `StatsPlugin`.